pkg/apis/crds/karpenter.k8s.aws_ec2nodeclasses.yaml

```diff
@@ -374,14 +385,22 @@ spec:
                 type: string
             required:
             - amiFamily
-            - role
             - securityGroupSelectorTerms
             - subnetSelectorTerms
             type: object
             x-kubernetes-validations:
             - message: amiSelectorTerms is required when amiFamily == 'Custom'
               rule: 'self.amiFamily == ''Custom'' ? self.amiSelectorTerms.size() !=
                 0 : true'
+            - message: expected at least one, got none, ['role', 'instanceProfile']
+              rule: has(self.role) || has(self.instanceProfile)
+            - message: role cannot be specified with instanceProfile
+              rule: '!(has(self.role) && has(self.instanceProfile))'
+            - message: changing from an unmanaged instance profile to a managed instance
```

Comment:
can we remove the concept of "managed" and just state "instanceProfile" vs "iamRole"